.circleci/config.yml

```diff
@@ -1,65 +1,72 @@
+      - run: curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.22.3/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
       - run:
           command: |
             sudo -E minikube start \
             --extra-config=apiserver.Admission.PluginNames="Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,GenericAdmissionWebhook,ResourceQuota" \
             --kubernetes-version=v1.7.5 --vm-driver=none
-      - run:
-          command: |
-            sudo -E minikube update-context
-      - run:
-          command: |
-            JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until sudo kubectl get nodes -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1; done
-      - run:
-          command: |
-            sudo -E kubectl cluster-info
-      - run:
-          command: |
-            ln -s ~/.kube/config pilot/platform/kube/
+      - run: sudo -E minikube update-context
+      - run: JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until sudo kubectl get nodes -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1; done
+      - run: sudo -E kubectl cluster-info
       - restore_cache:
           keys:
             - bazel-cache-{{ checksum "WORKSPACE" }}
-      - run: make test
+      - run: make build
       - save_cache:
           key: bazel-cache-{{ checksum "WORKSPACE" }}
           paths:
-            - /home/circleci/.cache/bazel
+            - /home/circleci/.cache/bazel      
       - run: make docker HUB=docker.io/rshriram TAG=test  # make docker images for everything
       - run: cd pilot; make e2etest HUB=docker.io/rshriram TAG=test TESTOPTS="-mixer=false -errorlogsdir=/home/circleci/logs"
       - run:
           command: |
             bash <(curl -s https://codecov.io/bash)
       - store_artifacts:
           path: /home/circleci/logs
-  test:
-    working_directory: /go/src/istio.io/istio
-    docker:
-      - image: circleci/golang:1.9
+
+  build:
+    machine: true
+    working_directory: ~/go/src/istio.io/istio
+    environment:
+      CHANGE_MINIKUBE_NONE_USER: true
     steps:
       - checkout
+      - run: curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.7.4/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+      - run: curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.22.3/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+      - run:
+          command: |
+            sudo -E minikube start \
+            --extra-config=apiserver.Admission.PluginNames="Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,GenericAdmissionWebhook,ResourceQuota" \
```

Comment:
Nit: might want to add `--extra-config=apiserver.Authorization.Mode=RBAC \` to make sure that all default RBAC rules are setup correctly (at least until it is enabled by default)